oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java

```diff
@@ -426,6 +434,9 @@ private void checkIfStopped() throws CommitFailedException {
 
     @Override
     public synchronized void run() {
+        if (!traverseNodesIfLaneNotPresentInIndex && !isIndexLanePresent()) {
```

Comment:
Could we restrict this check, so that it only checks if this is the very first run for this lane? I think there is a relatively simple way if that's the case: if there is no previous checkpoint for that indexing lane...

Reply:
I would probably do all the logic within the method isIndexLanePresent itself, so that there is only one method call here.

Reply:
But what if, new indexes gets introduced later on running instance. In that case, indexing for that lane will only start on bundle restart or instance restart.